solver/src/driver/solver_settlements.rs

```diff
@@ -44,28 +55,16 @@ pub fn merge_settlements(
     name: &'static str,
     settlements: Vec<Settlement>,
 ) -> SolverWithSettlements {
-    let mut settlements = settlements
-        .into_iter()
-        .map(|settlement| {
-            let objective_value = settlement.objective_value(prices);
-            RatedSettlement {
-                settlement,
-                objective_value,
-            }
-        })
-        .collect::<Vec<_>>();
-    settlements.sort_by(|a, b| b.objective_value.cmp(&a.objective_value));
-    if let Some(settlement) = merge_at_most_settlements(
-        max_merged_settlements,
-        settlements
-            .iter()
-            .map(|settlement| settlement.settlement.clone()),
-    ) {
-        let objective_value = settlement.objective_value(prices);
-        settlements.push(RatedSettlement {
-            settlement,
-            objective_value,
-        });
+    let mut settlements = settlements;
+
+    settlements.sort_by_cached_key(|a| -a.total_surplus(prices));
+
+    // I'd like a comment here: what is this doing? If there were N settlements, after
```

Comment:
It should take the top `max_merged_settlements` (in terms of surplus) and try to merge them into an additional solution which gets appended to the list of potential settlement (resulting in n+1 settlements). This is a short term way of increasing throughput for multiple disconnected settlements until we have parallel tx execution.

Maybe the sorting logic could go inside the `merge_at_most_settlements` function (to become part of the unit testing logic) even though it means `max_by_key` will be marginally less efficient as it will operate on unsorted data.

Reply:
I'm still unsure about it because I lack the understanding of what merged settlements are and why we're doing (is this written down in some issue?), so I won't try to change it in this PR.